x-pack/plugins/ml/public/application/components/ml_inference/components/advanced_configuration.tsx

```diff
@@ -0,0 +1,337 @@
+                      </EuiButtonEmpty>
+                    </EuiFlexItem>
+                    <EuiFlexItem grow={false}>
+                      {editInferenceConfig ? (
+                        <SaveChangesButton
+                          onClick={updateInferenceConfig}
+                          disabled={isInferenceConfigValid === false}
+                        />
+                      ) : null}
+                    </EuiFlexItem>
+                    <EuiFlexItem grow={false}>
+                      {editInferenceConfig ? (
+                        <EuiButtonEmpty size="xs" onClick={resetInferenceConfig}>
+                          {i18n.translate(
+                            'xpack.ml.trainedModels.content.indices.pipelines.addInferencePipelineModal.steps.advanced.resetInferenceConfigButton',
+                            { defaultMessage: 'Reset' }
+                          )}
+                        </EuiButtonEmpty>
+                      ) : null}
+                    </EuiFlexItem>
+                  </EuiFlexGroup>
+                }
+                error={inferenceConfigError ?? inferenceConfigModelTypeError}
+                isInvalid={
+                  inferenceConfigError !== undefined || inferenceConfigModelTypeError !== undefined
+                }
+              >
+                {editInferenceConfig ? (
+                  <CodeEditor
+                    height={300}
+                    languageId="json"
+                    options={{
+                      automaticLayout: true,
+                      lineNumbers: 'off',
+                      tabSize: 2,
+                    }}
+                    value={inferenceConfigString}
+                    onChange={handleInferenceConfigChange}
+                  />
+                ) : (
+                  <EuiCodeBlock isCopyable={true}>
+                    {JSON.stringify(inferenceConfig, null, 2)}
+                  </EuiCodeBlock>
+                )}
+              </EuiFormRow>
+            </EuiFlexItem>
+          </EuiFlexGroup>
+        </EuiFlexItem>
+        {/* FIELD MAP */}
+        <EuiFlexItem>
+          <EuiFlexGroup>
+            <EuiFlexItem grow={3}>
+              <EuiTitle size="s">
+                <h4>
+                  {i18n.translate(
+                    'xpack.ml.trainedModels.content.indices.pipelines.addInferencePipelineModal.steps.advanced.fieldMapTitle',
+                    { defaultMessage: 'Field map' }
+                  )}
+                </h4>
+              </EuiTitle>
+              <EuiSpacer size="m" />
+              <EuiText color="subdued" size="s">
+                <p>
+                  <FormattedMessage
+                    id="xpack.ml.trainedModels.content.indices.pipelines.addInferencePipelineModal.steps.advanced.fieldMapDescription"
+                    defaultMessage="Listed here are the fields the model expects. These are the fields the inference processor will default to if no {fieldMap} is specified."
```

Comment:
@valeriy42 - thanks for taking a look! 🙏 

I don't build this out since it defaults to what the model expects if this isn't added BUT if it's common for this to have to be set I could convert the array to a default field map 🤔  
Would you be up for providing context on what that should look like for the default?

Reply:
It's difficult to get the best UX here - I agree with @valeriy42 that you expect the form of the block on the left to be the same map format as the JSON block you end up editing. It's helpful to list out the fields that the model is expecting, but we don't want to provide some meaningless defaults like

<img width="1160" alt="image" src="https://github.com/elastic/kibana/assets/7405507/8d0a7839-c0fd-4d8a-b83c-5c4e48809096">

Is there some way the user can see a list of the fields the model expects (such as in a popover from the help text on the left), but then the JSON editor always shows the map format that is needed for `field_map`? That would also mean the edited configuration is shown, where currently after hitting edits you return to the list of expected field names.

Reply:
Thanks for the feedback, @peteharverson and @valeriy42 🙏 
This has been updated to keep the list hidden in a popover and now shows the expected json if the user wants to create the field_map. Updated in 64ba3ac4ec279eb78f5f8a97d01005991ae76219